components/scream/src/physics/shoc/shoc_functions_f90.hpp

```diff
@@ -16,9 +17,54 @@
 namespace scream {
 namespace shoc {
 
-///////////////////////////////////////////////////////////////////////////////
 
+///////////////////////////////////////////////////////////////////////////////
 // Converted subroutine helpers go here.
+struct SHOCGridData{
+static constexpr size_t NUM_ARRAYS = 4; 
+static constexpr size_t NUM_ARRAYS_i = 2; 
+
+    
+ // Inputs 
+ Int shcol, nlev, nlevi;
+ Real *zt_grid, *zi_grid, *pdel;
+
+ // In/out
+ Real *dz_zt, *dz_zi, *rho_zt;
+
+
+ SHOCGridData(Int shcol_, Int nevl_, Int nlevi_);
+ SHOCGridData(const SHOCGridData& rhs);
+ SHOCGridData& operator=(const SHOCGridData& rhs);
+ 
+ void init_ptrs();
+ //void transpose();
+
+ // Internals
+ Int m_shcol, m_nlev, m_nlevi, m_total, m_totali; 
+ std::vector<Real> m_data;
+ std::vector<Real> m_datai;
+
+ template <util::TransposeDirection::Enum D>
+ void transpose()
+ {
+   SHOCGridData d_trans(*this);
+   
+   // Transpose on the zt grid
+   util::transpose<D>(zt_grid, d_trans.zt_grid, shcol, nlev);
+   util::transpose<D>(dz_zt, d_trans.dz_zt, shcol, nlev);
+   util::transpose<D>(pdel, d_trans.pdel, shcol, nlev);
+   util::transpose<D>(rho_zt, d_trans.rho_zt, shcol, nlev);
+
+   // Transpose on the zi grid
+   util::transpose<D>(zi_grid, d_trans.zi_grid, shcol, nlevi);
+   util::transpose<D>(dz_zi, d_trans.dz_zi, shcol, nlevi);
+
+   *this = std::move(d_trans);
```

Comment:
Nice use of the move assignment. In order for this to work for me, I had to explicitly enable the default move assignment operator. Did you not have to?